packages/addon/src/channel_selector.ts

```diff
@@ -35,40 +40,95 @@ function fillChannels(data: ChannelDetails[], selected: string | null, messageCl
-      channels = details.channels
-      channelId = details.selected
-
-      const selectedColor = (channelId ? (channels.find(c => c.id == channelId)?.displayMetadata?.color) : null) ?? 'black'
-      logo.style.fill = selectedColor
+  const myPort = mc.port1;
+  myPort.start();
+  myPort.onmessage = ({data}) => {
+    console.debug("Received message: ", data);
+    switch(data.type){
+      case "iframeHandshake": {
+        collapse();
+        break;
+      }
+      case "iframeChannels": {
+        logo.removeEventListener("click", expand);
+        const {userChannels, selected} = data.payload as IframeChannelsPayload;
+        fillChannels(userChannels, selected, (channelStr) => {
+          myPort.postMessage({
+            type: "iframeChannelSelected",
+            payload: {
+              selected: channelStr || null
+            }
+          });
+          collapse();
+        });
+        const selectedChannel = userChannels.find((c) => c.id === selected);
+        logo.style.fill = selectedChannel?.displayMetadata?.color ?? "white";
+        logo.addEventListener("click", expand);
+        break;
+      }
     }
-  })
-
-  logo.addEventListener("click", () => {
-    fillChannels(channels, channelId, (id) => {
-      changeSize(false)
-      myPort.postMessage({ type: "SelectorMessageChoice", channelId: id })
-    })
-
-    // ask the parent container to increase the window size
-    changeSize(true)
-  })
+  };
+
+  parent.postMessage({
+    type: "iframeHello",
+    payload: {
+      initialCSS: {
+        width: `${8*4}px`,
+        height: `${8*5}px`,
+        right: "2px",
+        bottom: "2px",
+        zIndex: "1000",
+        "z-index": "1000",
+        position: "fixed"
+
+      }
+    }
+  }, "*", [mc.port2]);
+
+  const expand = () => {
+    document.body.setAttribute("data-expanded", "true");
+    myPort.postMessage({
+      type: "iframeRestyle",
+      payload: {
+        updatedCSS: {
+          width: `100%`,
+          height: `100%`,
+          top: 0,
+          left: 0,
+          zIndex: "1000",
+          "z-index": "1000",
+          position: "fixed"
+        }
+      }
+    });
+  }
 
+  const collapse = () => {
+    myPort.postMessage({
+      type: "iframeRestyle",
+      payload: {
+        updatedCSS: {
+          width: `${8*4}px`,
+          height: `${8*5}px`,
+          right: "2px",
+          bottom: "2px",
+          zIndex: "1000",
+          "z-index": "1000",
+          position: "fixed"
+        }
+      }
+    });
+
+    // If you immediately change to the logo, before the iframe has a chance to finish restyling,
+    // you see a flicker of a giant, colored logo.
+    // Here, we wait a negligible amount of time, and hope that the restyling has finished. This avoids the flicker.
+    // It's not a *good* idea, it's just the best available, since we don't know when the restyle finishes.
+    setTimeout(() => {
+      document.body.setAttribute("data-expanded", "false");
+    }, 15);
```

Comment:
Doh, do we need to add a response message to the restyle message so that the DA can indicate that it has applied the change? That's likely to add some lag however, so we might be just exchanging one artefact for another...

This is exactly the sort of issue that we need to explore and provide advice on handling so thank you for identifying it. We'll need to think though reliable solutions / designs that avoid this...